tests/test_libnetplan.py

```diff
@@ -293,6 +293,50 @@ def test_iter_ethernets_nameservers_and_domains(self):
         self.assertSetEqual(expected_domains, set(domain for domain in netdef.nameserver_search))
 
 
+class TestNetdefRouteIterator(TestBase):
+    def test_with_empty_routes(self):
+        state = state_from_yaml(self.confdir, '''network:
+  ethernets:
+    eth0: {}''')
+
+        netdef = next(netplan.netdef.NetDefinitionIterator(state, "ethernets"))
+        self.assertTrue(len([ip for ip in netdef.routes]) == 0)
+
+    def test_iter_routes(self):
+        state = state_from_yaml(self.confdir, '''network:
+  ethernets:
+    eth0:
+      routes:
+        - to: default
+          via: 192.168.0.1
+        - to: 1.2.3.0/24
+          via: 10.20.30.40
+          metric: 1000
+          table: 1000
+          from: 192.168.0.0/24
+        - to: 3.2.1.0/24
+          via: 10.20.30.40
+          metric: 1000
+          table: 1000
+          on-link: true
+          type: local
+          scope: host
+          mtu: 1500
+          congestion-window: 123
+          advertised-receive-window: 321
+          from: 192.168.0.0/24''')
+
+        netdef = next(netplan.netdef.NetDefinitionIterator(state, "ethernets"))
+        routes = [route for route in netdef.routes]
```

Comment:
```suggestion
        routes = list(netdef.routes)
```

suggestion (non-blocking): I guess this way it would be easier to read and according to https://www.curs-ml.com/post/fastest-way-to-convert-an-iterator-to-a-list it would even be faster. But OTOH this is a test case, so we can keep it as-is if you want to. The same applies to `ip in netdef.routes` (and others) above.

Reply:
oh, yeah I think at some point I was calling str() on route and ended up leaving the list comprehension there.